### DISCUSSION TITLE
Inverse Kinematics

### DISCUSSION THREAD

**Discussion by markusbuchholz** [Robotic-Decision-Making-Lab/alpha#83] [General]
Hei @evan-palmer,
 
For now, I will start again by conducting some research that includes manipulators. I plan to use the Bravo Alpha 5. My current research involves inverse kinematics and control, and I was wondering if you will be able to get IK running (in Alpha framework or other)

 I have tried to solve IK manually, in Bullet and, of course, in MoveIt 2 (my own approach), but I was unsuccessful. 
 
In the Alpha framework, you use MoveIt. Have you managed to solve the IK while running a C++ program?

Thanks

**Comment by evan-palmer** [comment]
Hey @markusbuchholz! When doing IK with the Alpha and the BlueROV, I used TPIK control, which worked quite nicely with the arm. Moving forward, I'm planning to use [Pinocchio](https://github.com/stack-of-tasks/pinocchio). I've used it with the Bravo, but haven't tested it with the Alpha yet.

**Reply by slopezba** [reply]
Hi, @markusbuchholz, I am Salvador from CIRTESU, Castellón in Spain, we meet in the last Martech at Palma, now I am working in the same field as you and experiencing the same issues as you with the inverse kinematics of the Alpha (or bravo 5) using MoveIt. Did you manage to solve the problem?
Thanks

**Reply by markusbuchholz** [reply]
Hi @slopezba , I have not solved this problem using Movit so in Python3 I use simple scipy for optimization and in C++ [ceres](http://ceres-solver.org/).

The C++ code I do not share now as it is a part of research project but Python code, documentiaon and all realted stuff you will find on my repo [here](https://github.com/markusbuchholz/ros2_moveit_manipulators?tab=readme-ov-file#lanuch-reach-alpha-5-underwater-applications)

I hope it helps!
Good luck.

**Reply by slopezba** [reply]
Thank you @markusbuchholz, I will take a look on the repo!
